src/core/namespace.c

```diff
@@ -1878,6 +1879,12 @@ static int apply_one_mount(
 
         assert(what);
 
+        if (m->mode == MOUNT_BIND && m->bind_mount_relabel) {
+                r = label_fix_full(AT_FDCWD, what, mount_entry_unprefixed_path(m), 0);
```

Comment:
as per coding style, could you pleas prefix the last argument with `/* flags= */` to indicate what the very opaque last param is?

Reply:
hmm, I think we should do this relabeling only when we create the dir, not when we use it.

also, this should really be moved inside relevant MOUNT_BIND/MOUNT_BIND_RECURSIVE branches of the switch statement above, and the if check not repeated here.

Reply:
actually, we don't really need a new bool field for this, the existing create_source_dir should be enough and it's set under the very same conditions already.

Reply:
I thought about reusing `create_source_dir` but then I decided for new field. Are you sure we want to do relabeling also in all future cases where `create_source_dir` would be set, dunno?

Reply:
please keep it simple, and I am pretty sure we anyway should only relabel this if we create it.